find-jruby-head-url-nokogiri.rb

```diff
@@ -12,7 +12,7 @@
 versions = Nokogiri::XML(xml).css('version').map(&:text)
 
 versions.delete('9000.dev-SNAPSHOT')
-most_recent = versions.max_by { |v| Gem::Version.new(v) }
+most_recent = versions.grep(/9\.4/).max_by { |v| Gem::Version.new(v) }
```

Comment:
Could you use an anchor here like `/^9\.4/` to avoid potentially wrong matches?

Reply:
It won't be here long enough to matter.

Reply:
Since I couldn't review before your merged, and this is my review, and I'm the main maintainer of setup-ruby & related repos I would like you to address this. I want to be sure we don't get some strange behavior because it matches e.g. `9.3.9.4`. And it's just more correct.

Reply:
There will never be a 9.3.9.4, nor a .9.4 for any previous release. In theory there could someday be a 10.0.9.4 or 10.9.4.0 but this patch will be gone before 10.0 is even released (within the next month).

> Since I couldn't review before your merged

It was necessary to expedite because we started pushing 10 snapshot builds and I knew users would start seeing those in setup-ruby before we're ready. Given the time-sensitive nature I opted to merge in a working patch. I'll add your refinement now.